code/game/objects/structures/extinguisher.dm

```diff
@@ -19,10 +19,24 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/extinguisher_cabinet, 29)
 		icon_state = "extinguisher_empty"
 	else
 		stored_extinguisher = new /obj/item/extinguisher(src)
+	register_context()
 
-/obj/structure/extinguisher_cabinet/examine(mob/user)
+/obj/structure/extinguisher_cabinet/add_context(atom/source, list/context, obj/item/held_item, mob/user)
 	. = ..()
-	. += span_notice("Right-click to [opened ? "close":"open"] it.")
+
+	if(isnull(held_item))
+		context[SCREENTIP_CONTEXT_RMB] = opened ? "Close" : "Open"
+		if(stored_extinguisher)
+			context[SCREENTIP_CONTEXT_LMB] = "Take extinguisher" //Yes, this shows whether or not it's open! Extinguishers are taken immediately on LMB click when closed
+
+	else if(held_item.tool_behaviour == TOOL_WRENCH)
+		if(!stored_extinguisher)
+			context[SCREENTIP_CONTEXT_LMB] = "Disassemble the cabinet"
+
+	else if(is_type(held_item, /obj/item/extinguisher))
+		if(!stored_extinguisher && opened)
+			context[SCREENTIP_CONTEXT_LMB] = "Insert extinguisher"
+
```

Comment:
```suggestion

	if(isnull(held_item))
		context[SCREENTIP_CONTEXT_RMB] = opened ? "Close" : "Open"
		if(stored_extinguisher)
			context[SCREENTIP_CONTEXT_LMB] = "Take extinguisher" //Yes, this shows whether or not it's open! Extinguishers are taken immediately on LMB click when closed
		return CONTEXTUAL_SCREENTIP_SET
	
	if(!stored_extinguisher)
		return NONE

	if(held_item.tool_behaviour == TOOL_WRENCH)
		context[SCREENTIP_CONTEXT_LMB] = "Disassemble cabinet"
		return CONTEXTUAL_SCREENTIP_SET

	if(istype(held_item, /obj/item/extinguisher) && opened)
		context[SCREENTIP_CONTEXT_LMB] = "Insert extinguisher"
		return CONTEXTUAL_SCREENTIP_SET
	
	return .

```

Reply:
Try this and let me know if it works

Reply:
Out of curiosity, what's it meant to do differently?

Reply:
No, it doesn't work. Wrench tips only appear when the extinguisher is in (which is when wrenches don't work), and holding an extinguisher has no tips either.

Reply:
By swapping TOOL_WRENCH and istype checks to be in an if(!stored_extinguisher) trait I can probably get them to appear and still be closer to what you want me to test, whatever it is

Reply:
> Out of curiosity, what's it meant to do differently?

It was psuedo code that I didn't test.  The whole point is to separate the `stored_extinguisher` var as an early return.

> By swapping TOOL_WRENCH and istype checks to be in an if(!stored_extinguisher) trait I can probably get them to appear and still be closer to what you want me to test, whatever it is

```
	if(stored_extinguisher)
		return NONE
```

Looks like I forgot to remove the `!` operator.  It should work now.  We are doing this to make the code cleaner.  Since the other `if()` loops had the same requirement, we move it outside the loop so we don't have to keep repeating it.

Make sense?

Reply:
Ahhhh, yea, makes sense. Keeping from even going thru the other loops since they get blocked by the extinguisher.

Reply:
Done and committed, feel free to open a new review if it's still looking off